### PR TITLE
MBS-10365 (V): Convert entity lists to react-table

### PR DESCRIPTION
**Pull request by reosarevok**
On top of https://github.com/metabrainz/musicbrainz-server/pull/1413

MBS-10365

* [ ] Artist
* [x] Event
* [x] Recording
* [x] Release
* [x] Release group

Artist (the only one left) is a bit more annoying since the component is currently being shared between search results and normal lists. We might want to split that one for now, until we can figure out how to easily do search results with react-table.

### CODE REVIEW

File: root/utility/tableColumns.js [hunk 1]
```diff
@@ -342,6 +353,31 @@ export function defineReleaseLabelsColumn(
...
+              title={l('Remove from merge')}
+              type="button"
+            />
+          </a>
+        </td>
+      ) : null;
+    },
+    Header: toMerge.length > 2 ? (
+      <th aria-label={l('Remove from merge')} style={{width: '1em'}} />
+    ) : null,
```
Comment (yvanzo) [inline root/utility/tableColumns.js]: Might be worth mentioning this code is being moved from  `root/components/RemoveFromMergeTableColumn.js` and `root/components/RemoveFromMergeTableHeader.js`, for example by inserting into these files a comment such as:
```js
// Converted to react-table at root/utility/tableColumns.js
```

Comment (reosarevok) [reply root/utility/tableColumns.js]: I'm removing these two files with the follow-up PRs anyway - one already with the artist PR, the other as soon as I get search working so I can get rid of ArtistListEntry :)

Comment (yvanzo) [reply root/utility/tableColumns.js]: Still, such comment or a note in commit message would make easier to track code in git history.

Comment (yvanzo) [reply root/utility/tableColumns.js]: By the way, why duplicating `RemoveFromMergeTableCell` for `Cell`?

Comment (reosarevok) [reply root/utility/tableColumns.js]: Because that file will also be deleted as soon as possible.

Comment (reosarevok) [reply root/utility/tableColumns.js]: Added that comment, anyway.

Comment (yvanzo) [reply root/utility/tableColumns.js]: For the record, added in https://github.com/metabrainz/musicbrainz-server/pull/1414/commits/291537028369ee52173e335118d31c918c7e8cc1.

**Comment by yvanzo** [comment]
You might want to cherry-pick https://github.com/yvanzo/musicbrainz-server/commit/2fd4a184ff5fe4a102b00be59d40d39b021db656 that patches `defineCountColumn` added in #1413.

File: flow-typed/npm/react-table_v7.x.x.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ declare module 'react-table' {
   declare export type ColumnInstance = {
     +className?: string,
     +getHeaderProps: (props?: {...}) => {...},
+    +headerProps?: {[attribute: string]: string},
```
[the remark this answers is not in the document]
Comment (yvanzo) [reply flow-typed/npm/react-table_v7.x.x.js]: Resolved in https://github.com/metabrainz/musicbrainz-server/pull/1414/commits/f52423726126a81f12e758c57be4670c8d01cc88#diff-acf4fb3aa97290aacc65193b4488f9b0R19

[the remark this answers is not in the document]
Comment (yvanzo) [reply flow-typed/npm/react-table_v7.x.x.js]: Resolved with https://github.com/metabrainz/musicbrainz-server/pull/1414/commits/f52423726126a81f12e758c57be4670c8d01cc88

**Pull request merged**